db/c.cc

```diff
@@ -417,6 +420,85 @@ struct rocksdb_filterpolicy_t : public FilterPolicy {
   const char* Name() const override { return (*name_)(state_); }
 };
 
+/* Event Listener */
+
+struct rocksdb_flushjobinfo_t {
+  const FlushJobInfo* rep;
+};
+
+struct rocksdb_event_listener_t : public EventListener {
+  void* state_;
+  void (*destructor_)(void*);
+  void (*on_flush_completed_)(void*, const rocksdb_flushjobinfo_t* info);
+
+  rocksdb_event_listener_t(void* state, void (*destructor)(void*))
+      : EventListener() {
+    state_ = state;
+    destructor_ = destructor;
```

Comment:
I find it a bit disconcerting that we don't assert these for non-nullness but it's like this throughout the FFI code...